controllers/reconcilers/kserve_networkpolicy_reconciler.go

```diff
@@ -0,0 +1,129 @@
+package reconcilers
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	kservev1beta1 "github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+	"github.com/opendatahub-io/odh-model-controller/controllers/comparators"
+	"github.com/opendatahub-io/odh-model-controller/controllers/components"
+	"github.com/opendatahub-io/odh-model-controller/controllers/processors"
+	"k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	networkPolicyName = "allow-from-openshift-monitoring-ns"
+)
+
+type KserveNetworkPolicyReconciler struct {
+	client               client.Client
+	scheme               *runtime.Scheme
+	networkPolicyHandler components.NetworkPolicyHandler
+	deltaProcessor       processors.DeltaProcessor
+}
+
+func NewKServeNetworkPolicyReconciler(client client.Client, scheme *runtime.Scheme) *KserveNetworkPolicyReconciler {
+	return &KserveNetworkPolicyReconciler{
+		client:               client,
+		scheme:               scheme,
+		networkPolicyHandler: components.NewNetworkPolicyHandler(client),
+		deltaProcessor:       processors.NewDeltaProcessor(),
+	}
+}
+
+func (r *KserveNetworkPolicyReconciler) Reconcile(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) error {
+
+	// Create Desired resource
+	desiredResource, err := r.createDesiredResource(isvc)
+	if err != nil {
+		return err
+	}
+
+	// Get Existing resource
+	existingResource, err := r.getExistingResource(ctx, log, isvc)
+	if err != nil {
+		return err
+	}
+
+	// Process Delta
+	if err = r.processDelta(ctx, log, desiredResource, existingResource); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *KserveNetworkPolicyReconciler) createDesiredResource(isvc *kservev1beta1.InferenceService) (*v1.NetworkPolicy, error) {
+	desiredNetworkPolicy := &v1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      networkPolicyName,
+			Namespace: isvc.Namespace,
+			Labels: map[string]string{
+				"app.kubernetes.io/version":               "release-v1.9",
```

Comment:
is it a good idea to hardcode it?

Reply:
With this PR I just mean to restructure code, even I am not sure why its hard coded here.

Reply:
okay, maybe we can open a issue to track it later.

Reply:
We already have a ticket to make hard-coded configuration as configurable. I am adding this one to that ticket itself.
https://github.com/opendatahub-io/odh-model-controller/issues/85